lambda/src/main/scala/pricemigrationengine/services/ZuoraLive.scala

```diff
@@ -208,11 +208,51 @@ object ZuoraLive {
           )
         }
 
+        override def applyAmendmentOrder(
+            subscription: ZuoraSubscription,
+            payload: ZuoraAmendmentOrderPayload
+        ): ZIO[Any, ZuoraOrderFailure, Unit] = {
+
+          // The existence of type_flush is explained in comment cada56ad.
+          // This is, for all intent and purpose a hack due to the way upickle deals with sealed traits
+          // and in a future change we will get rid of it, either by changing JSON library or coming up
+          // with the correct writers.
+
+          def type_flush(str: String): String = {
+            str
+              .replace(""""$type":"ZuoraAmendmentOrderPayloadOrderActionAdd",""", "")
+              .replace(""""$type":"ZuoraAmendmentOrderPayloadOrderActionRemove",""", "")
```

Comment:
Don't worry, we have to do this in Salesforce all the time as Apex has a lot of reserved keywords like `type`, `currency`, etc that causes issues with json parsing 😁